subgraphs/balancer-v2/src/common/getters.ts

```diff
@@ -0,0 +1,305 @@
+  let pool = new LiquidityPool(id);
+  let outputToken = getOrCreateToken(address);
+  let protocol = getOrCreateDex();
+
+  let wwPoolInstance = WeightedPool.bind(address);
+  let swapFees: BigInt = BigInt.fromI32(0);
+  let swapFeesCall = wwPoolInstance.try_getSwapFeePercentage();
+  if (!swapFeesCall.reverted) {
+    WeightedPoolTemplate.create(address);
+    swapFees = swapFeesCall.value;
+  } else {
+    let convergentCurvePool = ConvergentCurvePool.bind(address);
+    swapFeesCall = convergentCurvePool.try_percentFee();
+    if (!swapFeesCall.reverted) {
+      swapFees = swapFeesCall.value;
+    }
+  }
+
+  let feeInDecimals = scaleDown(swapFees, null);
+
+  let fee = new LiquidityPoolFee(id);
+  fee.feePercentage = feeInDecimals;
+  fee.feeType = LiquidityPoolFeeType.DYNAMIC_TRADING_FEE;
+  fee.save();
+
+  pool.name = outputToken.name;
+  pool.protocol = protocol.id;
+  pool.fees = [fee.id];
+  pool.totalValueLockedUSD = BIGDECIMAL_ZERO;
+  pool.cumulativeVolumeUSD = BIGDECIMAL_ZERO;
+  pool.inputTokenBalances = [BIGINT_ZERO];
+  pool.outputTokenSupply = BIGINT_ZERO;
+  pool.outputTokenPriceUSD = BIGDECIMAL_ZERO;
+  pool._totalSwapFee = BIGDECIMAL_ZERO;
+  pool._protocolGeneratedFee = BIGDECIMAL_ZERO;
+  pool._sideRevenueGeneratedFee = BIGDECIMAL_ZERO;
+  pool.rewardTokenEmissionsAmount = [BIGINT_ZERO];
+  pool.rewardTokenEmissionsUSD = [BIGDECIMAL_ZERO];
+  pool.createdBlockNumber = blockInfo.number;
+  pool.createdTimestamp = blockInfo.timestamp;
+  pool.outputToken = outputToken.id;
+  pool.save();
+
+  protocol._poolIds = protocol._poolIds.concat([pool.id]);
+  protocol.save();
+}
+
+export function getOrCreateFinancials(event: ethereum.Event): FinancialsDailySnapshot {
+  // Number of days since Unix epoch
+  let id: i64 = event.block.timestamp.toI64() / SECONDS_PER_DAY;
+
+  let financialMetrics = FinancialsDailySnapshot.load(id.toString());
+
+  if (!financialMetrics) {
+    financialMetrics = new FinancialsDailySnapshot(id.toString());
+    financialMetrics.protocol = getOrCreateDex().id;
+
+    financialMetrics.totalValueLockedUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyVolumeUSD = BIGDECIMAL_ZERO;
+    financialMetrics.cumulativeVolumeUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailySupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.cumulativeSupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.cumulativeProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyTotalRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.cumulativeTotalRevenueUSD = BIGDECIMAL_ZERO;
+
+    financialMetrics.save();
+  }
+  return financialMetrics;
+}
+
+export function getOrCreateDailyUsageMetricSnapshot(event: ethereum.Event): UsageMetricsDailySnapshot {
+  // Number of days since Unix epoch
+  let id: i64 = event.block.timestamp.toI64() / SECONDS_PER_DAY;
```

Comment:
The ID should follow this format `{ # of days since Unix epoch time }-{ HH: hour of the day }`

Reply:
Can you please share an example of another subgraph that is doing it this way? I took inspiration from Maker and Abracadabra

The reason I did not do it that way is that I couldn't find a way to get the hour based on only math. Probably I should just do something like:
```typescript
let dayId: i64 = event.block.timestamp.toI64() / SECONDS_PER_DAY;
let hourId: i64 = event.block.timestamp.toI64() / SECONDS_PER_HOUR;  

let hourlyMetricId = dayId.concat("-").concat(hourId)
```
Let me know if this makes sense to you :pray:

Reply:
Understood. Let's go with `{ Smart contract address of the pool }-{ # of hours since Unix epoch time }` then. Seems like this is how Uniswap is implementing it. I guess this makes querying a lot easier because you can sort this way. I will update the schema

Reply:
Okok cool, that's exactly how we are doing it :-D Here is the line of code: https://github.com/dOrgTech/subgraphs/blob/feat/balancer-v2/subgraphs/balancer-v2/src/common/getters.ts#L257